x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/apikey/QueryApiKeyRequest.java

```diff
@@ -13,32 +13,79 @@
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.core.Nullable;
 import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.search.searchafter.SearchAfterBuilder;
+import org.elasticsearch.search.sort.FieldSortBuilder;
 
 import java.io.IOException;
+import java.util.List;
 
 public final class QueryApiKeyRequest extends ActionRequest {
 
     @Nullable
     private final QueryBuilder queryBuilder;
+    private final int from;
+    private final int size;
+    @Nullable
+    private final List<FieldSortBuilder> fieldSortBuilders;
+    @Nullable
+    private final SearchAfterBuilder searchAfterBuilder;
     private boolean filterForCurrentUser;
 
     public QueryApiKeyRequest() {
         this((QueryBuilder) null);
     }
 
     public QueryApiKeyRequest(QueryBuilder queryBuilder) {
-        this.queryBuilder = queryBuilder;
+        this(queryBuilder, -1, -1, null, null);
```

Comment:
Default `size` is -1, which means it will use the default value from regular search, ie. `10`. We could bump this to a higher value, but I think it also has value to be consistent with the default search behaviour.